Source/RapyutaSimulationPlugins/Private/Core/RRNetworkPlayerController.cpp

```diff
@@ -247,11 +247,6 @@ void ARRNetworkPlayerController::ClientInitPawn_Implementation(AActor* InActor)
         robot->CreateROS2Interface();
         robot->ROS2Interface->Initialize(robot);
     }
-
```

Comment:
I needed to disable this to avoid initialize twice.
@Tadinu @larryng-rapyuta are those necessary?

Reply:
It looks like `URobotVehicleMovementComponent` itself is not replicated (not clear whether it needs), thus comes its instantiation here

Reply:
initmovecomponent is called at postinitialization method
https://github.com/rapyuta-robotics/RapyutaSimulationPlugins/blob/fe538bfcf873424f2e361954309aacfae4d7cd86/Source/RapyutaSimulationPlugins/Private/Robots/RRRobotBaseVehicle.cpp#L43

if you initialize here, it will initialize twice I think.

Reply:
Afaik, a robot is spawned at Server side, being initialized & get replicated down to all clients with _only its ActorComponents & members that are setup for replication_.
If MovementComponent is not setup for replicated, the robot at client side would not have it I suppose.

Reply:
it is set as replicated here
https://github.com/rapyuta-robotics/RapyutaSimulationPlugins/blob/fe538bfcf873424f2e361954309aacfae4d7cd86/Source/RapyutaSimulationPlugins/Public/Robots/RRRobotBaseVehicle.h#L64

Reply:
Ah that's only to replicate the pointer to its instance, which seems to appear as null on client side in my earlier try. Basically, to get an Actor component instance itself replicated, we need to invoke `SetIsReplicated(true)` like for [ROS2SimStateClient](https://github.com/rapyuta-robotics/RapyutaSimulationPlugins/blob/devel/Source/RapyutaSimulationPlugins/Private/Tools/RRROS2SimulationStateClient.cpp#L32).
But if you see it's already valid on client side, so probably meaning it's been invoked somewhere in parent classes.

Reply:
Ah that's only to replicate the pointer to its instance, which seems to appear as null on client side in my earlier try with ROS2SimStateClient. Basically, to get an Actor component instance itself replicated, we need to invoke `SetIsReplicated(true)` like for [ROS2SimStateClient](https://github.com/rapyuta-robotics/RapyutaSimulationPlugins/blob/devel/Source/RapyutaSimulationPlugins/Private/Tools/RRROS2SimulationStateClient.cpp#L32).
But if you see it's already valid on client side, so probably meaning it's been invoked somewhere in parent classes.

Reply:
got it any way, need more update later
- spawn movecomponent only on client side since currently client has authority on position
- add option to spawn movecomponent on both side to support server authority as well.

Reply:
Yeah, so I think for now it should be added to `ARRRobotBaseVehicle::PostInitializeComponents()`
```
if (IsNetMode(NM_Standalone))
{
   InitMoveComponent();
}
```
similar to ARRBaseRobot:: with ROS2Interface, which should be created by default for each robot in standalone server only.